src/JsonSchema/Pointer.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace JsonSchema;
+
+use JsonSchema\Exception\InvalidPointerException;
+use JsonSchema\Exception\ResourceNotFoundException;
+
+/**
+ * Resolve JSON Pointers (RFC 6901)
+ */
+class Pointer
```

Comment:
Alternatively, we could call this `PointerResolver` with a single method `resolvePointer($json, $pointer)`. I'd actually prefer this since the URI retriever's responsibility is loading schemas, not dealing with pointers.